python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -0,0 +1,168 @@
+
+    pass
+
+
+@dataclass
+class TATCredentials(auth.CredentialHelper):
+    ENV_VARIABLE: t.ClassVar[str] = "TX_TAT_CREDENTIALS"
+    FILE_NAME: t.ClassVar[str] = "~/.tx_tat_credentials"
+
+    username: str
+    password: str
+
+    @classmethod
+    def _from_str(cls, s: str) -> "TATCredentials":
+        user, _, passw = s.strip().partition(":")
+        return cls(user, passw)
+
+    def _are_valid(self) -> bool:
+        return bool(self.username and self.password)
+
+
+class TATSignalExchangeAPI(
+    auth.SignalExchangeWithAuth[CollaborationConfigBase, TATCredentials],
+    signal_exchange_api.SignalExchangeAPIWithSimpleUpdates[
+        CollaborationConfigBase, TATCheckpoint, TATSignalMetadata
+    ],
+):
+
+    def __init__(self, username, password) -> None:
+        super().__init__()
+        self.username = username
+        self.password = password
+
+    @staticmethod
+    def get_config_cls() -> t.Type[CollaborationConfigBase]:
+        return CollaborationConfigBase
+
+    @staticmethod
+    def get_checkpoint_cls() -> t.Type[TATCheckpoint]:
+        return TATCheckpoint
+
+    @staticmethod
+    def get_record_cls() -> type[TATSignalMetadata]:
+        return TATSignalMetadata
+
+    @classmethod
+    def for_collab(
+        cls,
+        collab: CollaborationConfigBase,
+        credentials: t.Optional[TATCredentials] = None,
+    ) -> "TATSignalExchangeAPI":
+        credentials = credentials or TATCredentials.get(cls)
+        return cls(
+            collab,
+            api.TATHashListAPI(
+                username=credentials.username, password=credentials.password
+            ),
+        )
+
+    def get_client(self) -> api.TATHashListAPI:
+        return api.TATHashListAPI(username=self.username, password=self.password)
+
+    def fetch_iter(
+        self,
+        _supported_signal_types: t.Sequence[t.Type[SignalType]],
+        checkpoint: t.Optional[TATCheckpoint],
+    ) -> t.Iterator[
+        state.FetchDelta[t.Tuple[str, str], TATSignalMetadata, TATCheckpoint]
+    ]:
+        """
+         The TAT Hash List returns a pre-signed URL to a JSON file containing a list of hashes.
+         As well as some metadata about the hash list and information about the hashes themselves including the requested ideology.
+
+        1. Fetch the hash list from the TAT API
+
+        2. Download the JSON file from the presigned URL
+
+        3. Read the JSON file and load into memory
+
+        4. Yield the delta mapping
+        """
+
+        client = self.get_client()
+        result = client.get_hash_list()
+
+        # Download the JSON file from the presigned URL
+        response = requests.get(result.file_url)
+        response.raise_for_status()
+
+        with tempfile.NamedTemporaryFile(delete=True) as f:
+            f.write(response.content)
+            f.close()
+
+            # Read the JSON file and load into memory
+            with open(f.name, "r") as file:
+                data = json.load(file)
```

Comment:
Lol, you don't need to write it to a file and then read it. Just use `loads` instead: https://docs.python.org/3/library/json.html